### PR TITLE
Improve debug env var handling

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: src/bin/miri.rs [hunk 1]
```diff
@@ -148,42 +150,31 @@ fn after_analysis<'a, 'tcx>(
...
+    // If MIRI_LOG is set and RUST_LOG is not, set RUST_LOG.
+    if let Ok(var) = env::var("MIRI_LOG") {
+        if env::var("RUST_LOG") == Err(env::VarError::NotPresent) {
+            // We try to be a bit clever here: If MIRI_LOG is just a single level
+            // used for everything, we only apply it to the parts of rustc that are
+            // CTFE-related.  Only if MIRI_LOG contains `module=level`, we just
+            // use the same value for RUST_LOG.
+            // This way, if you set `MIRI_LOG=trace`, you get only the right parts of
+            // rustc traced, but you can also do `MIRI_LOG=miri=trace,rustc_mir::interpret=debug`.
+            if var.contains('=') {
```
Comment (oli-obk) [inline src/bin/miri.rs]: I often specify `RUST_LOG=rustc_mir::interpret` without an `=`. In fact, I don't think I ever used `=`

Comment (RalfJung) [reply src/bin/miri.rs]: I didn't even know that's a thing. What does it do?

Comment (RalfJung) [reply src/bin/miri.rs]: I think I fixed this to properly recognize log levels.

Comment (oli-obk) [reply src/bin/miri.rs]: It enables all log levels for that module. So down to trace

**Pull request merged**